packages/react-pose/src/components/PoseElement.tsx

```diff
@@ -100,11 +108,11 @@ export class PoseElement extends React.PureComponent<PoseElementProps> {
     // If we're popping this element out from the DOM flow, build
     // and apply position: absolute styles that visually match the previous
     // location in the DOM
-    if (popFromFlow && this.ref && this.ref instanceof HTMLElement) {
+    if (popFromFlow && this.getRef() && this.getRef() instanceof HTMLElement) {
```

Comment:
Stick to `this.ref` - it doesn't matter where we got this from.

Reply:
Unfortunately, order matters with the ref. Check out [this example](https://codepen.io/jesstelford/pen/yKweKQ?&editors=0010):

```javascript
class SomeComponent extends React.Component {
  render() {
    return (
      <h1 ref={this.props.innerRef}>Hello World</h1>
    )
  }
}

class Posed extends React.Component {
  setInnerRef = () => {
    console.log('this.ref now points to <h1>');
  }
  setRef = () => {
    console.log('this.ref now points to <SomeComponent>');
  }
  render() {
    return (
      <SomeComponent ref={this.setRef} innerRef={this.setInnerRef} />
    )
  }
}

ReactDOM.render(
  <Posed />,
  document.getElementById('root')
);
```
Gives output:
```
this.ref now points to <h1>
this.ref now points to <SomeComponent>
```

Which brings us back to where we started (and the reason for this PR).

We could alias `this.hostRef` to `this.innerRef` without issue, so I'll make that change.

Reply:
I’m not sure I fully understand the problem on this one? this.ref is an internal reference for us that we expose via hostRef. It’s up to the user to ensure that’s passed correctly the the host element rather than the component (if it’s using forwardRef for instance both of those refs will be h1)

Reply:
Oh I see. It’s late 😁
Maybe we should only pass on hostRef for now as per docs, maybe innerRef if we’re srue it won’t be a problem

Reply:
Turns out trying to get to sleep is great at figuring this shit out (again). I see, actually this time. Proceed. 🙃

Reply:
Wait wouldn’t it be possible to check if elementtype was a string and only send ref if it was, otherwise it’s a component and sent hostRef and innerRef?

Reply:
Oh, of course! I'll give that a shot 👍

Reply:
Your tired self has good ideas too ;)